pkg/frontend/tgt/frontend.go

```diff
@@ -70,10 +74,18 @@ func (t *Tgt) Startup(rwu types.ReaderWriterUnmapperAt) error {
 
 	t.isUp = true
 
+	// If the engine failed during a snapshot, we may have left a frozen filesystem. This is a good opportunity to
+	// attempt to unfreeze it.
+	t.attemptUnfreezeFilesystem()
```

Comment:
Should we put this logic into [func (d *LonghornDevice) startScsiDevice(](https://github.com/longhorn/longhorn-engine/blob/a807f0fd6bfd9c4700f2c19808038e87a2ab814e/vendor/github.com/longhorn/go-iscsi-helper/longhorndev/dev.go#L122)

Reply:
While moving the logic into `go-iscsi-helper`, I realized that we should almost certainly execute it BEFORE `startFrontend`. If there are no available backends, we will never reach `startFrontend`. That's probably fine for the existing cleanup `startFrontend` might do (it'll strand a block device on the node), but it's NOT fine for the unfreeze case. We need to make sure the filesystem gets unfrozen, regardless of whether or not the engine can run.

As a result, I'm keeping the logic in `longhorn-engine`, but moving it to the `util` package so it can be executed earlier.